File: mega-gastro-poster-backend/src/main/java/com/megagastro/poster/model/Product.java
package com.megagastro.poster.model;

import lombok.Builder;
import lombok.Data;

@Builder
@Data
public class Product {

    private String id;
    private String name;
    private String url;
    private String imageUrl;
    private Double priceCurrent;
    private Double priceOriginal;
    private Integer discountPct;
    private ProductSource source;
    private String category;

}

File: mega-gastro-poster-frontend/types/product.ts
export type ProductSource = 'REMOTE' | 'CUSTOM';

export interface Product {
  id: string;
  name: string;
  url?: string | null;
  imageUrl?: string | null;
  priceCurrent: number;
  priceOriginal?: number | null;
  discountPct?: number | null;
  source: ProductSource;
  category?: string | null;
}


